src/SignAndMagnitudeInt.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+import JSBI from "jsbi";
+import {JsbiSupport} from "./JsbiSupport";
+
+/**
+ * Represents a signed, arbitrarily sized integer.
+ *
+ * BigInts cannot represent negative zero. This class should be used in situations where negative zero is
+ * a supported value, such as when decoding binary Ion Int/VarInt.
+ *
+ * http://amzn.github.io/ion-docs/docs/binary.html#uint-and-int-fields
+ *
+ */
+export default class SignAndMagnitudeInt {
```

Comment:
A comment earlier on in this PR said that this class is for internal use only, Should we prefix the name with `_`?

Reply:
We haven't been using underscore prefixes for internal classes thus far, just for member data / functions. Only [a handful of our classes](https://github.com/amzn/ion-js/blob/master/src/Ion.ts#L89-L97) are part of the public API.